internal/buff/read_test.go

```diff
@@ -34,7 +35,8 @@ func TestNext(t *testing.T) {
 	assert.Equal(t, uint8(0xa), r.MsgType)
 
 	expected := "cannot finish: unread data in buffer (message type: 0xa)"
-	assert.PanicsWithValue(t, expected, func() { r.Next(nil) })
+	assert.False(t, r.Next(nil))
+	assert.Equal(t, expected, r.Err.Error())
```

Comment:
A bit of a nit: this could be `assert.EqualError` https://pkg.go.dev/github.com/stretchr/testify/assert#EqualError